### ISSUE TITLE
JGit garbage collection is slow

### ISSUE THREAD

**Issue by Phillipus** [archimatetool/archi-modelrepository-plugin#108]
More of a heads up than bug report.

I hit "Refresh" (Pull) and the dialog was stuck for a long time. A debug trace showed that JGit was garbage collecting. Very slowly. I let it do its thing and it got there in the end. This was with JGit 4.11 and 5.3.0.

**Comment by janesser** [comment]
@Phillipus hi, "same procedure as every year" i tend to code a little. On the one hand getting a confined eclipse snap to work, which is no matter for here. Let me say, that i saw a lot good from you in the eclipse community.

On the other hand, i've been trying to poc my thoughts about getting around of jgit/egit, which have a lot of impact on my day2day. Especially because of slowness, see the other thread earlier: https://github.com/archimatetool/archi-modelrepository-plugin2/issues/11#issuecomment-2091118346

How has your idea evolved in the meanwhile - is coArchi2 the (only) way to go ? IIRC git would not be on the shortlist of sharing platforms with some mixed implications. 

My simple (maybe naive) proposal was started here: https://github.com/eclipse-egit/egit/issues/47
Basically wrapping around git-cli which is most optimized.

There is nothing but trying here:
https://github.com/janesser/archi-modelrepository-plugin/commit/347874f5f915b4eb2b0c9d20e573ea81fa95cd0d
https://github.com/janesser/archi-modelrepository-plugin/commit/58c1a442e5c18c8054c7380ae9956213fa53881b

Your reply would be very welcome. Cheers.

**Comment by Phillipus** [comment]
Hi, this issue is about the possibility of JGit being slow when garbage collecting. Let's keep it to that.

coArchi 2 will be continued see [Why coArchi 2?](https://github.com/archimatetool/archi-modelrepository-plugin2/wiki/Why-coArchi-2%3F) We don't use EGit at all, only JGit.

**Comment by janesser** [comment]
@Phillipus cool. i guess with a smaller set of files jgit gc will be less greedy.
let me know if i there is anything on coArchi 2 where a little contribution would help.

sorry for being the stubborn one;
anyway i will continue my kata attempting to show that a non-java integration of git deserves at least a little more consideration.

p.s. just looking through credential handling in coarchi and how the git-cli offers helpers. that is a descent puzzle to keep me busy.
